primary_beam_correction.py

```diff
@@ -207,63 +195,88 @@ def write_new_fits(pbc_image, path, outputFilename):
     hdr = images.header
     newhdr = hdr.copy()
     # add in required beam keywords
-    if newhdr['CLEANBMJ'] < 0:
-        # add in required beam keywords
-        newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
-        newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
-        newhdr['BPA'] = float(newhdr['HISTORY'][-2][55:])
+    try:
+        if newhdr['CLEANBMJ'] < 0:
+            # add in required beam keywords
+            newhdr['BMAJ'] = float(newhdr['HISTORY'][-2][20:30])
+            newhdr['BMIN'] = float(newhdr['HISTORY'][-2][36:48])
+            newhdr['BPA'] = float(newhdr['HISTORY'][-2][55:])
 
-        newhdr['CTYPE3'] = 'FREQ'
-        newhdr['NAXIS3'] = 1
-        newhdr['CDELT3'] = 1.0
-        newhdr['CRVAL3'] = 1.284e9
-    else:
-        # add in required beam keywords
-        newhdr['BMAJ'] = newhdr['CLEANBMJ']
-        newhdr['BMIN'] = newhdr['CLEANBMN']
-        newhdr['BPA'] = newhdr['CLEANBPA']
-        # change the frequency plane keywords, we don't want multiple frequency
-        newhdr['CTYPE3'] = 'FREQ'
-        newhdr['NAXIS3'] = 1
-        newhdr['CDELT3'] = 1.0
+            newhdr['CTYPE3'] = 'FREQ'
+            newhdr['NAXIS3'] = 1
+            newhdr['CDELT3'] = 1.0
+            newhdr['CRVAL3'] = 1.284e9
```

Comment:
Will `CRVAL3` always be 1.284e9?

Reply:
I have fixed it. Does the CRVAL3 change if we still using the same receiver?

Reply:
CRVAL3 will change depending on the instrumental setup. In particular you can be using L-band in narrowband mode which can be tuned to any frequency over the 843MHz range of L-band and therefore CRVAL3 will be different if the center frequency of the observation isn't 1.284e9. You can get the center frequency from the original FITS image.